### ISSUE TITLE
Using a Select input in a cell

### ISSUE THREAD

**Issue by linojon** [mui-org/material-ui-x#2516]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current Behavior 😯

Not sure if this is a bug or a feature request. I have been struggling to add a Select input to a cell. It doesnt retain the input values. I hacked it by adding renderCell, renderEditCell, and onChange methods which helps but still isnt right. Here'e a codesandbox demo

https://codesandbox.io/s/divine-tdd-qrxu3?file=/src/Demo.tsx

### Steps to Reproduce 🕹

Steps:

1.
2.
3.
4.

**Comment by DanailH** [comment]
Hi, @linojon thanks for raising this. Did you check the column types we support OOTB https://material-ui.com/components/data-grid/columns/#column-types. 

It sounds like you are looking for the `singleSelect` column type, is that the case?

**Comment by linojon** [comment]
yes i discovered that after submitting this. i didnt notice that because i dont think its mentioned nor examples on the Editing page, and there's no reason to declare a column as 'singleSelect' in view mode.

But this issue remains if i want to substitute your built in input type with my own input component, even a simple replacement for the default singleSelect. I see the "Custom Edit Component" topic but that i think now requires Pro for the api? (not tagged as such on the page). Are we saying that renderEditCell itself is also only a Pro feature? (not tagged as such on the Columns page).

**Comment by DanailH** [comment]
@linojon sorry for the delay. No, you can still use the `renderEditCell` to override our OOTB input. the params that `renderEditCell` receives are of type `GridRenderEditCellParams ` which also includes the `GridApi`.

Hope that helps.

**Comment by missbruni** [comment]
@DanailH 
I have been changing my columns to try and use the default implementation for this select, using type: 'singleSelect' and 'valueOptions' but we are unable to do because we need to set the valueOptions based on another cell value in the same row. And the valueOptions do not give you access to the row values. Any suggestions we can do to be able to use the singleSelect in a way that we can get access to a value from another row cell.

**Comment by m4theushw** [comment]
@DanailH Based on https://github.com/mui-org/material-ui-x/issues/2516#issuecomment-918043682, I think we should support passing a function to `valueOptions`. What do you think?
